Sources/SignalProducer.swift

```diff
@@ -471,13 +585,33 @@ extension SignalProducer {
 	/// - returns: A signal producer that applies signal's operator to every
 	///            created signal.
 	public func lift<U, F>(_ transform: @escaping (Signal<Value, Error>) -> Signal<U, F>) -> SignalProducer<U, F> {
-		return SignalProducer<U, F> { () -> SignalProducer<U, F>.Instance in
+		return SignalProducer<U, F>(SignalCore {
 			// Transform the `Signal`, and pass through the `didCreate` side effect and
 			// the interruptHandle.
-			let instance = self.producer.builder()
-			return SignalProducer<U, F>.Instance(producedSignal: transform(instance.producedSignal),
-			                                     observerDidSetup: instance.observerDidSetup,
-			                                     interruptHandle: instance.interruptHandle)
+			let receipt = self.core.make()
+			return .init(signal: transform(receipt.signal),
+			             observerDidSetup: receipt.observerDidSetup,
+			             interruptHandle: receipt.interruptHandle)
+			})
+	}
+
+	private func lift<U, F, V, G>(leftFirst: Bool, _ transform: @escaping (Signal<Value, Error>) -> (Signal<U, F>) -> Signal<V, G>) -> (SignalProducer<U, F>) -> SignalProducer<V, G> {
+		return { otherProducer in
+			return SignalProducer<V, G>(SignalCore {
+				let left = self.core.make()
+				let right = otherProducer.core.make()
+
+				return .init(signal: transform(left.signal)(right.signal),
+				             observerDidSetup: {
```

Comment:
Could we put all the arguments on their own line? 🙈 

```swift
return .init(
    signal: transform(left.signal),
    observerDidSetup: {
        ...
    },
    ...
)
```

It's minor, but I think it has readability benefits. (Including outdenting the block body.)